rust/chains/hyperlane-fuel/src/conversions.rs

```diff
@@ -0,0 +1,56 @@
+use hyperlane_core::H256;
+
+/// Conversion from a fuel type to H256 primitive.
+pub trait FuelIntoH256 {
```

Comment:
Due to how traits work there really is not a good way to do this any other way. Because you have to implement traits where they are define _or_ where the structure is defined and we are defining this exclusively for remote types, that means we need to import all the things it will be implemented for where the trait is defined.

The downside to making a generic one is that wherever we put that, it would have to import every library we need to support down the road. We might move the macro to be in hyperlane-core, but hyperlane-core SHOULD NOT depend on every library like fuel and ethers and more as we grow. So yeah. That is why.